### PR TITLE
Pass on 'other props' to the <Search /> component

### PR DESCRIPTION
**Pull request by samuelmale**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

We are using `PatientTable` as an extension, we have the requirement of passing a number of callbacks eg. `onKeyDown`. 

## Screenshots

*None.*

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Comment by samuelmale** [comment]
cc: @manuelroemer / @donaldkibet

File: packages/esm-patient-list-app/src/patient-table/patient-table.component.tsx [hunk 1]
```diff
@@ -94,9 +96,10 @@ const PatientTable: React.FC<PatientTableProps> = ({ patients, columns, search,
           placeholder={search.placeHolder}
           onChange={(evnt) => handleSearch(evnt.target.value)}
           className={styles.searchOverrides}
-          value={search.currentSearchTerm}
+          defaultValue={search.currentSearchTerm}
```
[the remark this answers is not in the document]
Comment (manuelroemer) [reply packages/esm-patient-list-app/src/patient-table/patient-table.component.tsx:99]: Not really sure about this - does this cause problems in the "normal" patient-list-app, i.e. where the component is not used as an extension?

What were the issues with the `value` for you?

[the remark this answers is not in the document]
Comment (manuelroemer) [reply packages/esm-patient-list-app/src/patient-table/patient-table.component.tsx:99]: > This component isn't actually used in the Patient list app. I believe it's only us (OHRI team) using this component.

Ahh, thanks for clarifying this, I wasn't aware. In that case everything's a little relaxed. Then let's just keep the `defaultValue` in the PR!

**Comment by brandones** [comment]
Agreed, @manuelroemer . Personally my concern is about the sort of ad-hoc way that extensions are being "adapted" for re-use elsewhere. I worry that they might grow into "bad abstractions." Having our abstractions centrally located might help head that off and make it easier to keep track of them. Does that make sense to you?

**Pull request merged**